presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/framework/BenchmarkRunner.java

```diff
@@ -16,72 +16,78 @@
 import com.facebook.airlift.event.client.EventClient;
 import com.facebook.presto.benchmark.event.BenchmarkPhaseEvent;
 import com.facebook.presto.benchmark.event.BenchmarkSuiteEvent;
-import com.facebook.presto.benchmark.executor.PhaseExecutor;
-import com.facebook.presto.benchmark.executor.PhaseExecutorFactory;
 import com.facebook.presto.benchmark.source.BenchmarkSuiteSupplier;
 import com.google.common.collect.ImmutableSet;
 import com.google.inject.Inject;
 
 import javax.annotation.PostConstruct;
 
-import java.util.List;
 import java.util.Set;
 
 import static com.facebook.presto.benchmark.event.BenchmarkPhaseEvent.Status.FAILED;
 import static com.facebook.presto.benchmark.event.BenchmarkPhaseEvent.Status.SUCCEEDED;
+import static com.facebook.presto.benchmark.framework.ExecutionStrategy.CONCURRENT;
+import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class BenchmarkRunner
 {
     private final BenchmarkSuiteSupplier benchmarkSuiteSupplier;
-    private final PhaseExecutorFactory phaseExecutorFactory;
+    private final ConcurrentPhaseExecutor concurrentPhaseExecutor;
```

Comment:
So basically `BenchmarkRunner` will have as many executors are there are execution strategies?

Reply:
Yes, and there will only be 2 in the foreseeable future. `CONCURRENT` and `STREAM`